generators/chipyard/src/main/scala/IOBinders.scala

```diff
@@ -366,13 +368,14 @@ class WithTiedOffDebug extends OverrideIOBinder({
   (system: HasPeripheryDebugModuleImp) => {
     val (psdPort, resetctrlOpt, debugPortOpt, ioCells) =
       AddIOCells.debug(system.psd, system.resetctrl, system.debug)(system.p)
-    val harnessFn = (th: chipyard.TestHarness) => {
+    val harnessFn = (th: HasHarnessUtils) => {
       Debug.tieoffDebug(debugPortOpt, resetctrlOpt, Some(psdPort))(system.p)
       // tieoffDebug doesn't actually tie everything off :/
       debugPortOpt.foreach { d =>
-        d.clockeddmi.foreach({ cdmi => cdmi.dmi.req.bits := DontCare; cdmi.dmiClock := th.clock })
+        // TODO: Using harness clock/reset will be incorrect when systemClock =/= harnessClock
```

Comment:
So it turns out the way we drive the debug module clocks is pretty wrong, debug.clock should not be punched off-chip to be driven.
We should fix the WithSimDebug and WithTiedOffDebug IOBinders in another PR.

Reply:
Sounds good.

Reply:
I think there is probably some confusion between the DMI_DTM signals going off chip (which is only intended as a simulation config) and the JTAG_DTM version which has a pretty clear set of signals that should go off-chip (and is intended to be actually implemented in the real world). Perhaps this confusion is carried through to the IOBinders?

Reply:
Yeah that is accurate. I think the problem is that we are plumbing through the entire `DebugIO` bundle, instead of just `DebugIO.systemjtag` or `DebugIO.clockeddmi`. 

`DebugIO.clock` appears to require a clock which is synchronous to the Tilelink buses (https://github.com/chipsalliance/rocket-chip/blob/3176ed81c2bfa1a24622c511a1e4b0777bc9377c/src/main/scala/devices/debug/Debug.scala#L1775-L1780), so it should not be generated offchip. I feel like we should be calling `Debug.connectClockAndReset` somewhere within the chip...